File: _etc/C#/2579 계단 오르기.cs
using System;

class Program
{
    static void Main() {
        string s = Console.ReadLine();
        int n = int.Parse(s);
        
        int[] arr = new int[301];
        int[] dp = new int[301];
        
        for(int i = 1; i <= n; i++) {
            string s2 = Console.ReadLine();
            int num = int.Parse(s2);
            arr[i] = num;
        }
        
        dp[0] = 0;
        dp[1] = arr[1];
        dp[2] = arr[1] + arr[2];
        
        for(int i = 3; i <= n; i++) {
            dp[i] = Math.Max(arr[i] + arr[i-1] + dp[i-3], arr[i] + dp[i-2]);
        }
        
        Console.WriteLine(dp[n]);
    }
}

// http://boj.kr/1d16a0d76ffd4c799c473f1e540faca1

File: mvc1/Python/2579 계단오르기.py
n = int(input())


arr = [0 for i in range(301)]
dp = [0 for i in range(301)]

for i in range(1, n+1) :
    arr[i] = int(input())

dp[0] = 0
dp[1] = arr[1]
dp[2] = arr[1] + arr[2]

for i in range(3, n+1) :
    dp[i] = max(arr[i] + arr[i-1] + dp[i-3], arr[i] + dp[i-2])

print(dp[n])

# http://boj.kr/c02ec863af7c4767b89b1919adf97220
